### PR TITLE
downloads file content only if user requests

### PR DESCRIPTION
**Pull request by prudhvije**
downloads file content only if user requests.Storing pulled results in map so redundant requests won't be made to ES.

### CODE REVIEW

**Comment by prudhvije** [comment]
This will close #76

**Comment by ScrapCodes** [comment]
Can you also add tests for elastic search queries ?

**Comment by prudhvije** [comment]
if correct file is being downloaded or not??

**Comment by prudhvije** [comment]
I mean filecontents?

**Comment by prudhvije** [comment]
Added Tests for elastic search queries.

File: plugins/idea/pluginTests/src/test/scala/com/imaginea/betterdocs/ESQueryJsonSuit.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+package com.imaginea.betterdocs
+
+import org.scalatest.{BeforeAndAfterAll, FunSuite}
+import scala.collection.JavaConversions._
+
+class ESQueryJsonSuit extends FunSuite with BeforeAndAfterAll{
+  val imortsSet = Set("java.util.Map")
+  val size = 10
+  val actualJSON = new JSONUtils().getESQueryJson(imortsSet, size)
+  println(actualJSON)
```
Comment (ScrapCodes) [inline plugins/idea/pluginTests/src/test/scala/com/imaginea/betterdocs/ESQueryJsonSuit.scala]: remove printlns

**Comment by prudhvije** [comment]
Renamed file and updated tests.

**Comment by prudhvije** [comment]
Fixed scalastyle errors.

**Comment by ScrapCodes** [comment]
merged

**Pull request closed without merging**